web3_proxy/src/frontend/users/payment_stripe.rs

```diff
@@ -0,0 +1,240 @@
+
+    let db_replica = app.db_replica().context("Getting database connection")?;
+
+    // Filter by user ...
+    let receipts = stripe_increase_balance_receipt::Entity::find()
+        .filter(increase_on_chain_balance_receipt::Column::DepositToUserId.eq(user.id))
+        .all(db_replica.as_ref())
+        .await?;
+
+    // Return the response, all except the user ...
+    let response = json!({
+        "user": Address::from_slice(&user.address),
+        "deposits": receipts,
+    });
+
+    Ok(Json(response).into_response())
+}
+
+/// `POST /user/balance/stripe/` -- Process a confirmed txid to update a user's balance.
+#[debug_handler]
+pub async fn user_balance_stripe_post(
+    Extension(app): Extension<Arc<Web3ProxyApp>>,
+    ip: Option<InsecureClientIp>,
+    Path(mut params): Path<HashMap<String, String>>,
+    bearer: Option<TypedHeader<Authorization<Bearer>>>,
+) -> Web3ProxyResponse {
+    // rate limit by bearer token **OR** IP address
+    let (authorization, _semaphore) = if let Some(TypedHeader(Authorization(bearer))) = bearer {
+        let (_, semaphore) = app.bearer_is_authorized(bearer).await?;
+
+        // TODO: is handling this as internal fine?
+        let authorization = Web3ProxyAuthorization::internal(app.db_conn())?;
+
+        (authorization, Some(semaphore))
+    } else if let Some(InsecureClientIp(ip)) = ip {
+        let authorization = login_is_authorized(&app, ip).await?;
+
+        (authorization, None)
+    } else {
+        return Err(Web3ProxyError::AccessDenied);
+    };
+
+    // Get the payload, and the header
+    let payload = params.remove("data").ok_or(Web3ProxyError::BadRequest(
+        "You have not provided a 'data' for the Stripe payload".into(),
+    ))?;
+
+    // TODO Get this from the header
+    let signature = params
+        .remove("STRIPE_SIGNATURE")
+        .ok_or(Web3ProxyError::BadRequest(
+            "You have not provided a 'STRIPE_SIGNATURE' for the Stripe payload".into(),
+        ))?;
+
+    // Now parse the payload and signature
+    // TODO: Move env variable elsewhere
+    let event = Webhook::construct_event(
+        &payload,
+        &signature,
+        std::env::var("STRIPE_API_KEY")
+            .expect("STRIPE_API_KEY must be set.")
+            .as_str(),
+    )
+    .context(Web3ProxyError::BadRequest(
+        "Could not parse the stripe webhook request!".into(),
+    ))?;
+
+    let intent = match event.data.object {
+        stripe::EventObject::PaymentIntent(intent) => intent,
+        _ => return Ok("Received irrelevant webhook".into_response()),
+    };
+    debug!("Found PaymentIntent Event: {:?}", intent);
+
+    if intent.status.to_string() != "succeeded" {
+        return Ok("Received Webhook".into_response());
+    }
+
+    let db_conn = app.db_conn().context("query_user_stats needs a db")?;
+
+    if stripe_increase_balance_receipt::Entity::find()
+        .filter(stripe_increase_balance_receipt::Column::Id.eq(intent.id.as_str()))
```

Comment:
Wrong column. Should be StripePaymentIntendId

Reply:
fixed